### PR TITLE
fix: test mode flag

### PR DESCRIPTION
**Pull request by frosso**
Fixes https://github.com/Automattic/woocommerce-payments/issues/2450

#### Changes proposed in this Pull Request

The test mode flag is not being retrieved correctly, causing the store to be in a permanent "test mode" state after saving the settings.

#### Testing instructions

I tested this in my `wpsandbox.me` site (not locally), so that the "dev mode" check would be falsy.
You can also add `add_filter( 'wcpay_dev_mode', '__return_false' );` on your site to set dev mode to false.

- Go to the settings page
- Enable test mode
- Save the settings
- Refresh the page
- Notice the "test mode" notice being added
- Disable test mode
- Save the settings
- Refresh the page
- The "test mode" notice is removed
- Checkout allows test cards

Before: it wasn't allowing checkout with live card numbers
Screen Shot 2021-07-21 at 10 31 05 AM

After: it allows checkout with live card numbers
Screen Shot 2021-07-21 at 10 31 14 AM

-------------------

- [x] Added changelog entry (or does not apply)
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

### CODE REVIEW

File: includes/admin/class-wc-rest-payments-settings-controller.php [hunk 1]
```diff
@@ -177,7 +177,7 @@ public function get_settings(): WP_REST_Response {
 				'available_payment_method_ids'      => $this->wcpay_gateway->get_upe_available_payment_methods(),
 				'is_wcpay_enabled'                  => $this->wcpay_gateway->is_enabled(),
 				'is_manual_capture_enabled'         => 'yes' === $this->wcpay_gateway->get_option( 'manual_capture' ),
-				'is_test_mode_enabled'              => 'yes' === $this->wcpay_gateway->is_in_test_mode(),
+				'is_test_mode_enabled'              => $this->wcpay_gateway->is_in_test_mode(),
```
Comment (frosso) [inline includes/admin/class-wc-rest-payments-settings-controller.php:180]: The whole issue is that the `is_in_test_mode` method does not return a string, but a boolean

**Comment by harriswong** [comment]
👀

File: tests/unit/admin/test-class-wc-rest-payments-settings-controller.php [hunk 2]
```diff
@@ -130,6 +130,28 @@ public function test_get_settings_returns_available_payment_method_ids() {
 		);
 	}
 
+	public function test_get_settings_request_returns_test_mode_flag() {
+		$this->gateway->update_option( 'test_mode', 'yes' );
+		add_filter(
+			'wcpay_dev_mode',
+			function () {
+				return true;
```
Comment (harriswong) [inline tests/unit/admin/test-class-wc-rest-payments-settings-controller.php:138]: Should dev mode be false in order to test both test_mode as yes|no?

Comment (frosso) [reply tests/unit/admin/test-class-wc-rest-payments-settings-controller.php:137]: Sure thing - I made some changes in 9eb35c3c to test all the combinations

**Review by harriswong — APPROVED** [review]
👍  LGTM

**Pull request merged**